bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java

```diff
@@ -594,4 +594,11 @@ static Field adjustField(Field field, @Nullable Field existingField) {
     // no adjustment
     return field;
   }
+
+  public static String prepareQueryForLog(String query, int maxLength) {
+    String noNewLinesQuery = query.replace("\n", "");
+    return noNewLinesQuery.length() > maxLength
+        ? noNewLinesQuery.substring(0, maxLength) + /* ellipsis */ '\u2026'
```

Comment:
can consider logging the truncated part length.

Reply:
The full query is in the log. We put the first 50 characters, and it is just for the query plan